### ISSUE TITLE
Instanced scenes share their metadata if the saved PackedScene contains saved metadata

### ISSUE THREAD

**Issue by mrcdk** [godotengine/godot#29189]
**Godot version:**

Godot 3.0.6, Godot 3.1.1 and latest Godot master https://github.com/godotengine/godot/commit/0d8f1ba6a9182d1f1af39097e2165fc9807377f5

**OS/device including version:**

Windows 10

**Issue description:**

~~If the Control's children are instances of the same scene the children of the same scene will share the same metadata.~~

Instanced scenes share their metadata if the saved `PackedScene` contains saved metadata in the scene's root node.

This is problematic because master now saves metadata for each control and some controls like the `TabContainer` and its `TabContainer.set_tab_*()` methods use the child's metadata to change some internal values.

~~As far as I can tell this doesn't happen with Node, Node2D or Spatial instances.~~

**Steps to reproduce:**
Open the attached MRP and run the Main* scenes. Only `MainControl` will show:
``` 
True
False
False
True
```
The expected result (and the result you will get with all others) is:
```
False
False
False
False
```

Notice that `InstancedControl.tscn` and `AnotherControlInstanced.tscn` contains saved metadata while the other *.tscn don't. If you add metadata manually the problem will appear in those instances.

**Minimal reproduction project:**

[test_meta_instances.zip](https://github.com/godotengine/godot/files/3220183/test_meta_instances.zip)

**Comment by mrcdk** [comment]
Tested on Godot 2.1.5 and it doesn't happen there so it's probably a 3.0+ issue.

**Comment by mrcdk** [comment]
Okay, my bad, I've found the issue. It doesn't happen in 3.0.x or 3.1.x It only happens on master.

Master now saves the following metadata automatically in each PackedScene.:
```
__meta__ = {
"_edit_use_anchors_": true
}
```

And that's what's causing the problem in the first place. I manually added metadata to one of the instanced *.tscn files and I could replicate the issue in 3.0+ but not in 2.1.5

I made a quick fix here https://github.com/godotengine/godot/blob/0d8f1ba6a9182d1f1af39097e2165fc9807377f5/core/object.cpp#L437-L443 and modified `metadata = p_value;` to `metadata =  p_value.duplicate(true);` (deep duplicate) which fixed the issue in master but because it doesn't happen in 2.1.5 I think that's not the correct fix.

**Comment by Calinou** [comment]
I noticed an issue tangential to this on commit https://github.com/godotengine/godot/commit/b030834b6384d5f57e5fc63db21b643adc86b3da, not sure if it's related. Metadata such as edit locking or Position2D gizmo extents doesn't get saved when you change it on an instanced scene in the scene tree dock. It will be lost when you reload the scene.

**Comment by KoBeWi** [comment]
Might be related/duplicate of #20648

**Comment by akien-mga** [comment]
Fixed by #32416.
